### PR TITLE
spg Chapter 11 - Not quite working dynamic proxy example

### PR DESCRIPTION
**Pull request by SimonGross**
I was trying to get this working, but I ran out of time.

My goal here was to have dynamic proxies on two different interfaces, but because I couldn't get the proxies to  work on their own I wasn't able to set up anything more complicated that what I've got here.

Despite trying to follow along with the example in the book, it doesn't appear to work correctly:
if I call `editorAccess.setArticleText` and then `editorAccess.getArticleText`, it will return null.

### CODE REVIEW

File: spg/src/com/spg/chapter11/NewsWebsite.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	/**
+	 * A nice, non-controversial article
+	 */
+	public static void niceArticle() {
+		Article article = new OpinionArticle();
+
+		// Editor writes the article
+		Article editorAccess = getArticleProxy( article, new ArticleEditorHandler( article ) );
+		editorAccess.setTitle( "Man's best friend" );
+		editorAccess.setArticleText( "I like puppies." );
```
Comment (a-panayotopoulos) [inline spg/src/com/spg/chapter11/NewsWebsite.java]: :dog:

File: spg/src/com/spg/chapter11/handler/ArticleEditorHandler.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		super();
+		this.article = article;
+	}
+
+	@Override
+	public Object invoke( Object proxy, Method method, Object[] args ) throws IllegalAccessException {
+		
+		try {
+			if ( method.getName().startsWith( "get" ) ) {
+				method.invoke( article, args );
```
Comment (a-panayotopoulos) [inline spg/src/com/spg/chapter11/handler/ArticleEditorHandler.java]: Here's your problem. You need `return method.invoke( article, args );` here and everywhere else.

Comment (SimonGross) [inline spg/src/com/spg/chapter11/handler/ArticleEditorHandler.java]: :+1: Thanks, that fixed the issue.

**Comment by a-panayotopoulos** [comment]
Neat. Do the tests pass now?

**Comment by SimonGross** [comment]
Yeah, the tests work. I didn't think it was worth showing at the meeting because it was just a bunch of console output but they do work.

**Comment by a-panayotopoulos** [comment]
:shipit:

**Pull request merged**